tensorflow_recommenders/tasks/retrieval.py

```diff
@@ -36,7 +36,12 @@ class Retrieval(tf.keras.layers.Layer, base.Task):
   This task defines models that facilitate efficient retrieval of candidates
   from large corpora by maintaining a two-tower, factorized structure: separate
   query and candidate representation towers, joined at the top via a lightweight
-  scoring function.
+  scoring function. Unlike tfrs.tasks.Ranking, this retrieval task assumes your
```

Comment:
If we're expanding this, perhaps we should mention why we impose this factorized structure here? Something along the lines of:

This two-tower factorized structure is required for retrieval tasks, because it allows the candidate representations to be computed ahead of time and stored in a fast retrieval index (such as `tfrs.layers.factorized_top_k.ScaNN`). Consequently, this tasks assumes that your.... (the rest of your edit follows).

Reply:
Yes this is really useful context, will add and make sure it flows.